mcs/class/corlib/System.Runtime.Remoting.Messaging/MonoMethodMessage.cs

```diff
@@ -68,8 +68,38 @@ internal class MonoMethodMessage : IMethodCallMessage, IMethodReturnMessage, IIn
 		internal static String CallContextKey = "__CallContext";
 		internal static String UriKey           = "__Uri";
 
-		[MethodImplAttribute(MethodImplOptions.InternalCall)]
-		internal extern void InitMessage (MonoMethod method, object [] out_args);
+		internal void InitMessage (MonoMethod method, object [] out_args)
+		{
+			this.method = method;
+			ParameterInfo[] paramInfo = method.GetParametersInternal ();
+			int param_count = paramInfo.Length;
```

Comment:
Yes, those reflection icalls are also on our list to rewrite in managed.  (Alternately I could surface a struct from C right here with just `Name`, `IsOut` and `IsByRef` for each param, but I was under the impression that this isn't one of the perf-critical portions of the runtime)